src/components/ButtonDropdown.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+  <div ref="root" :class="['button-dropdown', { open: isOpen }]">
+    <button 
+       :class="['button-dropdown-button', buttonClass]" 
+      @click="toggleVisibility"
+    >
+      <span>
+        {{ buttonText }}
+      </span>
+    </button>
+    <div v-show="isOpen" class="button-dropdown-content">
+      <slot></slot>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'button-dropdown',
+  props: ['buttonText', 'buttonClass'],
+  data() {
+    return {
+      isOpen: false
+    };
+  },
+  methods: {
+    toggleVisibility() {
+      this.isOpen = !this.isOpen;
+    },
+    handleWindowClick(event) {
+      if (this.$refs.root.contains(event.target)) {
+        return;
+      } else if (this.isOpen) {
+        this.isOpen = false;
+      }
+    }
+  },
+  mounted() {
+    window.addEventListener('click', this.handleWindowClick);
```

Comment:
IE10 will blow up with this but IE10 is .88% of browser share...maybe this is ok? ...